spec/requests/api/instances_spec.rb

```diff
@@ -274,4 +274,82 @@ def update_raw_power_state(state, *instances)
       expect_result_resources_to_include_hrefs("results", :instances_list)
     end
   end
+
+  context "Instance shelve action" do
+    it "responds not found for an invalid instance" do
+      api_basic_authorize action_identifier(:instances, :shelve)
+
+      run_post(invalid_instance_url, gen_request(:shelve))
+
+      expect_resource_not_found
+    end
+
+    it "responds forbidden for an invalid instance without appropriate role" do
+      api_basic_authorize
+
+      run_post(invalid_instance_url, gen_request(:shelve))
+
+      expect_request_forbidden
+    end
+
+    it "shelves a powered off instance" do
+      api_basic_authorize action_identifier(:instances, :shelve)
+      update_raw_power_state("SHUTOFF", instance)
+
+      run_post(instance_url, gen_request(:shelve))
+
+      expect_single_action_result(:success => true, :message => 'shelving', :href => :instance_url)
+    end
+
+    it "shelves a suspended instance" do
+      api_basic_authorize action_identifier(:instances, :shelve)
+      update_raw_power_state("SUSPENDED", instance)
+
+      run_post(instance_url, gen_request(:shelve))
+
+      expect_single_action_result(:success => true, :message => 'shelving', :href => :instance_url)
+    end
+
+    it "shelves a paused instance" do
+      api_basic_authorize action_identifier(:instances, :shelve)
+      update_raw_power_state("PAUSED", instance)
+
+      run_post(instance_url, gen_request(:shelve))
+
+      expect_single_action_result(:success => true, :message => 'shelving', :href => :instance_url)
+    end
+
+    it "cannot shelve a shelved instance" do
+      api_basic_authorize action_identifier(:instances, :shelve)
+      update_raw_power_state("SHELVED", instance)
+
+      run_post(instance_url, gen_request(:shelve))
+
+      expect_single_action_result(
+        :success => false,
+        :message => "The VM can't be shelved, current state has to be powered on, off, suspended or paused",
+        :href    => :instance_url
+      )
+    end
+
+    it "shelves an instance" do
+      api_basic_authorize action_identifier(:instances, :shelve)
+
+      run_post(instance_url, gen_request(:shelve))
+
+      expect_single_action_result(:success => true,
+                                  :message => "shelving",
+                                  :href    => :instance_url,
+                                  :task    => true)
+    end
+
+    it "shelves multiple instances" do
+      api_basic_authorize action_identifier(:instances, :shelve)
+
+      run_post(instances_url, gen_request(:shelve, nil, instance1_url, instance2_url))
```

Comment:
Hm....this was done for all the other multiple action specs, as per the VMs API. The second arg is data - would we want to send any?